src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1867,21 +1867,7 @@ private static bool ShouldReportNullableAssignment(TypeWithAnnotations type, Nul
                     GetParameterAsDiagnosticArgument(parameterOpt),
                     GetContainingSymbolAsDiagnosticArgument(parameterOpt));
 
-                if (targetType.Type.IsPossiblyNullableReferenceTypeTypeParameter())
```

Comment:
I don't think it's correct to remove this. We may want to refine the condition here, though. IIRC `[DisallowNull]` on unconstrained generic types tends to require a little special handling.

```cs
class C<T>
{
    void M1()
    {
        T? t = default;
        M2(t); // warning. Update state to maybe-null.
        M2(t); // no warning.
        t.ToString(); // warn again. Update state to not-null.
    }
    void M2(T t) { }
    
    void M3()
    {
        T? t = default;
        M2(t); // warning. Update state to not-null.
        M2(t); // no warning.
        t.ToString(); // no warning.
    }
    void M4([DisallowNull] T t) { }
}
```

Reply:
@cston Do you have any thoughts? From offline discussion with Rikki, we feel that changing the state to not-null during invocation would probably make sense (we still give a safety warning and reduce cascading diagnostics).

Reply:
> From offline discussion with Rikki, we feel that changing the state to not-null during invocation would probably make sense (we still give a safety warning and reduce cascading diagnostics).

That sounds reasonable to me.